cloud/pkg/synccontroller/synccontroller.go

```diff
@@ -149,20 +155,34 @@ func (sctl *SyncController) deleteObjectSyncs() {
 	if err != nil {
 		klog.Errorf("Failed to list all the ObjectSyncs: %v", err)
 	}
-	for _, sync := range syncs {
-		nodeName := getNodeName(sync.Name)
-		isGarbage, err := sctl.checkObjectSync(sync)
-		if err != nil {
-			klog.Errorf("failed to check ObjectSync outdated, %s", err)
-			continue
-		}
-		if isGarbage {
-			klog.Infof("ObjectSync %s will be deleted since node %s has been deleted", sync.Name, nodeName)
-			err = sctl.crdclient.ReliablesyncsV1alpha1().ObjectSyncs(sync.Namespace).Delete(context.Background(), sync.Name, *metav1.NewDeleteOptions(0))
+	retryTime := 0
+	errSync := make([]*v1alpha1.ObjectSync, 0)
+	for retryTime < maxRetries && len(syncs) != 0 { // If an error occurred while deleting ObjectSync, try again
```

Comment:
Would it be a problem to retry the loop from the beginning every time?
Should it be:
```go
for _, sync := range syncs {
    // retry sync item...
}
```

Reply:
My previous idea was to put the error sync in a separate `errsync` list, and after the end of this cycle, copy the `errsync` list to the `syncs` queue and repeat the cycle. But thanks for the reminder. It would be more concise to complete the retry directly when processing each sync.

Reply:
By the way, the err in ` Verify vendor` and `Multiple build` CI doesn't seem to be related to this PR, I see you fixed them in #6045, right? @WillardHu

Reply:
> By the way, the err in ` Verify vendor` and `Multiple build` CI doesn't seem to be related to this PR, I see you fixed them in #6045, right? @WillardHu

Yes, you can wait for the PR #6045 merged, or also fix join_windows.go and edgecoreinstaller.go.